File: Week 6/pset6/sentimental-credit/credit.py
from cs50 import get_int

def main():
    # Prompt card number
    card_number = get_int("Number: ")

    if validate_card(card_number):
        print(get_card_system(card_number))
    else:
        print("INVALID")

    return 0

def get_card_system(card_number):
    # 1. Check the length of card number
    # 2. Check the first two digits
    start = card_number
    while start > 100:
        start = start // 10
    if ((start == 34) or (start == 37)) and len(str(card_number)) == 15:
        card_system = "AMEX"
    elif (start >= 51) and (start < 56) and len(str(card_number)) == 16:
        card_system = "MASTERCARD"
    elif (start >= 40) and (start < 50) and len(str(card_number)) >= 13 and len(str(card_number)) < 17:
        card_system = "VISA"
    else:
        card_system = "INVALID"
    return card_system

def validate_card(card_number):
    first_sum = 0
    second_sum = 0
    copy_card_number = card_number
    second_card_number = card_number
    x = 0
    i = 0
    while (i < 15):
        first_sum += copy_card_number % 10
        copy_card_number = copy_card_number // 100
        i += 2
    i = 0
    while (i < 15):
        second_card_number = second_card_number // 10
        x = (second_card_number % 10) * 2
        if x > 9:
            second_sum += x % 10
            x = x // 10
            second_sum += x % 10
            x = 0
            second_card_number = second_card_number // 10
            i += 2
        else:
            second_sum += x
            second_card_number = second_card_number // 10
            i += 2
    if (first_sum + second_sum) % 10 == 0:
        return True
    else:
        return False

main()

File: Week 1/Problem Set 1/credit/credit.c
#include <cs50.h>
#include <stdio.h>

void credit_card_system(long card_number)
{
    if (card_number < 3800000000000000 && card_number >= 3400000000000000)
    {
        printf("AMEX\n");
    }
    else if (card_number < 5600000000000000 && card_number >= 5100000000000000)
    {
        printf("MasterCard\n");
    }
    else if (card_number < 5000000000000000 && card_number >= 4000000000000000)
    {
        printf("Visa\n");
    }
    else {
        printf("Invalid card number.\n");
    }
}

void validate_card(long card_number)
{
    int first_sum = 0;
    long copy_card_number = card_number;
    long second_card_number = card_number;
    int x = 0;
    for (int i = 0; i < 15; i = i + 2)
    {
        first_sum += copy_card_number % 10;
        copy_card_number = copy_card_number / 100;
        printf("copy_card_number: %li, fist_sum: %i\n", copy_card_number, first_sum);
    }
    int second_sum = 0;
    for (int i = 0; i < 15; i = i + 2)
    {
        second_card_number = second_card_number / 10;
        x = (second_card_number % 10) * 2;
        if (x > 9)
        {
            second_sum += x % 10;
            x = x / 10;
            second_sum += x % 10;
            x = 0;
            second_card_number = second_card_number / 10;
        }
        else
        {
        second_sum += x;
        second_card_number = second_card_number / 10;
        }
        printf("second_sum: %i\n", second_sum);
    }
    if ((first_sum + second_sum) % 10 == 0)
    {
        printf("VALID\n");
    }
    else
    {
        printf("INVALID\n");
    }
}

int main(void)
{
    long card_number = get_long("Number: ");
    credit_card_system(card_number);
    validate_card(card_number);
}